### PR TITLE
[16.0][FIX] pos_order_to_sale_order

### PR DESCRIPTION
**Pull request by geomer198**
At creating SO from POS, description computed by default with adding customer note from POS.

### CODE REVIEW

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi. Don't understand that PR. 
I tested on runboat with the 16.0 module : 
- add a pos order line
- add a comment
- create a draft sale order

the comment is present in the name of the sale order line.

image

**Comment by geomer198** [comment]
> Hi. Don't understand that PR. I tested on runboat with the 16.0 module :
> 
>     * add a pos order line
> 
>     * add a comment
> 
>     * create a draft sale order
> 
> 
> the comment is present in the name of the sale order line.
> 
> image


In default module behavior product internal reference in sol is not displayed. Now products Internal reference is displayed.

**Comment by legalsylvain** [comment]
Thanks for the explanation ! 

Well, the problem IMO is that this PR introduce a useless field at sale.order.line level that will take space in the DB. We should consider another design.

I guess call ``_get_sale_order_line_multiline_description_sale`` in the prepare function should do the job. Don't you think ?

**Comment by geomer198** [comment]
> `_get_sale_order_line_multiline_description_sale`

Hello @legalsylvain ! This function (_get_sale_order_line_multiline_description_sale) is called after created SOL, so at preparing value I think better solution adding new field for saving customer_note.
For the applying `_get_sale_order_line_multiline_description_sale` function then need create so and sol without preparing value. I think this solution is not good( 
What do you think about it?

**Comment by legalsylvain** [comment]
I think you can pass information by context. It could make the job. 
but creating a useless field in database is an issue IMO.

**Comment by ivs-cetmix** [comment]
> I think you can pass information by context. It could make the job. but creating a useless field in database is an issue IMO.

Hi @legalsylvain we have approached this problem from another angle) Could you have a look please?

**Comment by legalsylvain** [comment]
Hi. I'm on holidays the next week. I'll take a look when I'm back to the office.

**Comment by ivs-cetmix** [comment]
> Hi. I'm on holidays the next week. I'll take a look when I'm back to the office.

Thank you! Have a nice holidays)

File: pos_order_to_sale_order/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -1,6 +1,7 @@
 * Go to Point Of Sale / Configuration / Point of Sale
 * Check the box 'Create Sale Orders'
 * Select the desired default behaviour
+* Select the SO Line Name mode
```
Comment (ivs-cetmix) [inline pos_order_to_sale_order/readme/CONFIGURE.rst]: Select Sales Order Lina name composition mode

File: pos_order_to_sale_order/readme/newsfragments/2928.bigfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Sale order line description computed by default with adding Customer note.
```
Comment (ivs-cetmix) [inline pos_order_to_sale_order/readme/newsfragments/2928.bigfix.rst]: This is not a bug. A feature is being added.
Add readable and clear feature description please

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]

File: pos_order_to_sale_order/models/res_config_settings.py [hunk 3]
```diff
@@ -23,3 +23,13 @@ class ResConfigSettings(models.TransientModel):
...
         related="pos_config_id.iface_create_invoiced_sale_order", readonly=False
     )
+
+    sol_name_mode = fields.Selection(
+        selection=[
+            ("product_pos", "Product name + POS Comment"),
+            ("multiline", "Sale Multiline Description"),
+        ],
+        string="SO Line Name Mode",
+        default="product_pos",
```
Comment (ivs-cetmix) [inline pos_order_to_sale_order/models/res_config_settings.py:33]: What happens if user will remove selection value? So no option is selected

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]

**Comment by ivs-cetmix** [comment]
> Hi. I'm on holidays the next week. I'll take a look when I'm back to the office.

Hi @legalsylvain sorry for bothering you again) Would be great if you could have a look at this

File: pos_order_to_sale_order/models/sale_order_line.py [hunk 4]
```diff
@@ -10,16 +10,22 @@ class SaleOrderLine(models.Model):
...
+            and self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("pos_order_to_sale_order.sol_name_mode", "product_pos")
+            == "product_pos"
+        ):
+            product = self.env["product.product"].browse(order_line_data["product_id"])
+            product_name = product.name
+            product_name += "\n" + order_line_data["customer_note"]
+            vals.update(name=product_name)
+        return vals
```
Comment (legalsylvain) [inline pos_order_to_sale_order/models/sale_order_line.py:31]: Maybe I'm wrong, but it looks like if the previous condition is wrong, the vals returned doesn't contain any "name" value. Did I missed something ?

Comment (geomer198) [reply pos_order_to_sale_order/models/sale_order_line.py:31]: > Maybe I'm wrong, but it looks like if the previous condition is wrong, the vals returned doesn't contain any "name" value. Did I missed something ?

Thank you for you question! 
Name field is auto complete after creating SOL. And this solution for the set custom value for SOL description.

**Comment by legalsylvain** [comment]
Hi @geomer198.

> Hello @legalsylvain ! This function (_get_sale_order_line_multiline_description_sale) is called after created SOL, so at preparing value I think better solution adding new field for saving customer_note.

I don't think so.  ``_get_sale_order_line_multiline_description_sale`` is called by the ``_compute_name`` function, before the call during the call of create.

I made an alternative implementation here #1080.

Thanks for you review.

**Comment by geomer198** [comment]
> Hi @geomer198.
> 
> > Hello @legalsylvain ! This function (_get_sale_order_line_multiline_description_sale) is called after created SOL, so at preparing value I think better solution adding new field for saving customer_note.
> 
> I don't think so. `_get_sale_order_line_multiline_description_sale` is called by the `_compute_name` function, before the call during the call of create.
> 
> I made an alternative implementation here #1080.
> 
> Thanks for you review.

Thank you for your solution!

**Comment by geomer198** [comment]
> Hi @geomer198.
> 
> > Hello @legalsylvain ! This function (_get_sale_order_line_multiline_description_sale) is called after created SOL, so at preparing value I think better solution adding new field for saving customer_note.
> 
> I don't think so. `_get_sale_order_line_multiline_description_sale` is called by the `_compute_name` function, before the call during the call of create.
> 
> I made an alternative implementation here #1080.
> 
> Thanks for you review.

Thanks for your solution.
This solution looks better than my.  I close my PR in favor of your implementation.

**Pull request closed without merging**